### DISCUSSION TITLE
Get started

### DISCUSSION THREAD

**Discussion by sweco-secfbr** [RealEstateCore/rec#129] [Q&A] [answered]
If I as a realestaet owner want to build a REC representation for one of my buildings. What tool can I use to create triplets to defind the building? Where do I store the data?  How do I know what relations predicate exists in REC? How can I create unique UUIDs for my entities?

**Comment by hammar** [comment, marked as answer]
Hi,

Thanks for reaching out on this and your other questions/issues! I'm a little under water at the moment but I hope to respond to you on these matters later in the week (Thursday or Friday).

As pertains to this particular question: there are at least two approaches to how you might utilize an OWL ontology:

1. You might represent your building data (both the structure of the building itself and the sensor data / command data relating to operating it) as RDF triples, using any RDF-capable tooling -- of which there are plenty. Triple stores for storing such data include OntoText GraphDB, Amazon Neptune, Stardog, or an open source tool such as Jena. Programming APIs for ingesting and emitting RDF include the Jena API, Apache RDF4J (both Java), DotNetRDF (.NET), RDFLib (Python), etc.
2. You might use the OWL ontology as a data schema for data that is stored in a non-RDF system, e.g., CosmosDB, Neo4J, a relational database, etc. In some cases there exist solutions that are more-or-less turn-key, whereas in other cases you may need to write your own storage/mapping solution.

I don't quite understand the question on "relations predicate" -- but if you want to read the REC docs, see https://doc.realestatecore.io/. 3.3 is the closest to where we are now going, but still a bit tentative. 

URI/URN minting can be a non-trivial challenge. As you allude to, you might typically combine a namespace component (e.g., "https://data.mycompany.com/buildingData/" with a UUID; or you might prefer to have human-readable URIs (e.g. using a generally agreed-upon name for a building, or room, or other feature, instead of a UUID). The benefit of using URIs over URNs is that the former can be resolved and retrieved using standard DNS resolvers and HTTP servers/clients, which is sometimes a powerful machinery to utilize.

Best,

Karl
